### PR TITLE
Update scheduled sampling harness

### PR DESCRIPTION
**Pull request by dschodt-lanl**
This merge request updates the scheduled sampling training harness to use a cosine with warmup learning rate scheduler.

### CODE REVIEW

**Review by hickmank — CHANGES_REQUESTED** [review]
The CosineWithWarmup LR-scheduler needs to be stepped every batch. So you need to modify the "training step" function in `torch_training_utils.py`. See `torch_training_utils.train_LRsched_loderunner_epoch` and the function called therein for guidance.

**Comment by hickmank** [comment]
> This PR looks good, just a couple of pointers:
> 
>     1. Please add a description, what updates are you making here? (I know you are switching the learning rate scheduler)... The description should answer "why does this PR exist?", and include whatever else you want.
> 
>     2. I noticed you switched to the debug queue, is that intentional? This is not a problem, just something you want to be aware of.
> 
> 
> None of these issues are critical, but these are things you should keep in mind!

I agree with adding a description (not too detailed) to the PR.

File: applications/harnesses/chicoma_lsc_loderunner_scheduled/train_density_LodeRunner_scheduled.py [hunk 1]
```diff
@@ -277,7 +284,7 @@
         )
 
         # Increment LR scheduler
-        stepLRsched.step()
+        LRsched.step()
```
[the remark this answers is not in the document]
Comment (galegozi) [reply applications/harnesses/chicoma_lsc_loderunner_scheduled/train_density_LodeRunner_scheduled.py:287]: I think you should be able to get rid of this line completely, pretty sure the cosine scheduler steps automatically

[the remark this answers is not in the document]
Comment (dschodt-lanl) [reply applications/harnesses/chicoma_lsc_loderunner_scheduled/train_density_LodeRunner_scheduled.py]: I've removed the `LRsched.step()` in the harness and added the appropriate update step in `torch_training_utils.train_scheduled_loderunner_epoch()`

File: src/yoke/torch_training_utils.py [hunk 2]
```diff
@@ -1493,6 +1496,9 @@ def train_scheduled_loderunner_epoch(
                 scheduled_prob=scheduled_prob
             )
 
+            # Increment the learning-rate scheduler
+            LRsched.step()
```
Comment (hickmank) [inline src/yoke/torch_training_utils.py:1500]: Awesome! Yep, this works.

**Review by hickmank — APPROVED** [review]
Looks good. You're coverage is decreasing but only a very little bit. Can you write one unittest somewhere so the coverage doesn't decrease?

**Comment by dschodt-lanl** [comment]
> Looks good. You're coverage is decreasing but only a very little bit. Can you write one unittest somewhere so the coverage doesn't decrease?

I added a missing unit test for `cli.add_scheduled_sampling_args()`.

**Review by hickmank — APPROVED** [review]
Thanks. This looks good.
